src/main/java/org/elasticsearch/index/query/CommonTermsQueryBuilder.java

```diff
@@ -217,4 +292,136 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
+            if (analyzerObj == null) {
+                throw new IllegalArgumentException("No analyzer found for [" + analyzer + "]");
+            }
+        }
+        
+        Occur highFreqOccur = highFreqOperator.equals(Operator.AND) ? Occur.MUST : Occur.SHOULD;
+        Occur lowFreqOccur = lowFreqOperator.equals(Operator.AND) ? Occur.MUST : Occur.SHOULD;
+
+        ExtendedCommonTermsQuery commonsQuery = new ExtendedCommonTermsQuery(highFreqOccur, lowFreqOccur, cutoffFrequency, disableCoord, mapper);
+        Query query = parseQueryString(commonsQuery, text, field, analyzerObj, lowFreqMinimumShouldMatch, highFreqMinimumShouldMatch);
+
+        query.setBoost(boost);
+        if (queryName != null) {
+            parseContext.addNamedQuery(queryName, query);
+        }
+        return query;
+    }
+    
+    public static final Query parseQueryString(ExtendedCommonTermsQuery query, String queryString, String field, Analyzer analyzer, 
+                                         String lowFreqMinimumShouldMatch, String highFreqMinimumShouldMatch) throws IOException {
+        // Logic similar to QueryParser#getFieldQuery
+        int count = 0;
+        try (TokenStream source = analyzer.tokenStream(field, queryString.toString())) {
+            source.reset();
+            CharTermAttribute termAtt = source.addAttribute(CharTermAttribute.class);
+            BytesRefBuilder builder = new BytesRefBuilder();
+            while (source.incrementToken()) {
+                // UTF-8
+                builder.copyChars(termAtt);
+                query.add(new Term(field, builder.toBytesRef()));
+                count++;
+            }
+        }
+
+        if (count == 0) {
+            return null;
+        }
+        query.setLowFreqMinimumNumberShouldMatch(lowFreqMinimumShouldMatch);
+        query.setHighFreqMinimumNumberShouldMatch(highFreqMinimumShouldMatch);
+        return query;
+    }
+
+    @Override
+    public QueryValidationException validate() {
+        QueryValidationException validationException = null;
+        if (this.fieldName == null || this.fieldName.isEmpty()) {
+            validationException = QueryValidationException.addValidationError("field name cannot be null or empty.", validationException);
+        }
+        if (this.text == null || this.text.isEmpty()) {
+            validationException = QueryValidationException.addValidationError("No text specified for text query", validationException);
+        }
+        return validationException;
+    }
+
+    @Override
+    public CommonTermsQueryBuilder readFrom(StreamInput in) throws IOException {
+        CommonTermsQueryBuilder commonTermsQueryBuilder = new CommonTermsQueryBuilder(in.readString(), in.readString());
+        commonTermsQueryBuilder.highFreqOperator = in.readString().equals("AND") ? Operator.AND : Operator.OR;
+        commonTermsQueryBuilder.lowFreqOperator = in.readString().equals("AND") ? Operator.AND : Operator.OR;
```

Comment:
Maybe try and sync with what Isabel is doing on her Simple query string PR, she has the same operator there too, maybe we can share it throughout different queries